### PR TITLE
Feat/new frontend fixes

### PR DESCRIPTION
**Pull request by u22520016**
Hey guys...I tried to fix everything correctly this time...I'm just sending a reminder of what I was supposed to change...please make sure nothing breaks code already in develop...I manually copied and pasted the changes, so there shouldn't be any issues.  (Also I know there is an overflow issue on the events page.  It is where the height of the upcoming events cards is too small and the content inside overflows...I tried to look at it and I know the problem.  When the width of the screen changes, the height of those cards changes as well (so I think it has something to do with the aspect ratio). Could I ask the overflow queen @MonikaTheiss if you know how to fix it?)

Hey guys...this is a big PR. I am going to try to remember what I fixed.

1. I fixed all App bars to look the same and all "sub" pages' app bars are now smaller and have a color. It fits some standards for app bars.
2. I redid the whole home page.
3. You can now view the list of event invites on the events page (top right corner) **(this is new since last time)**
4. I added a way to directly go to the search page from the home page to make the home page more interesting
6. I changed the event page a bit. I removed the top space when no events were upcoming or joined. I also changed the tab buttons a bit. I also removed the empty space at the bottom
7. I removed the reviews and genres from the game library page. I also added "Metacritic" to the Metacritic score and changed the position.
8. I added padding to the search bar and also added a clear button to clear search input text
9. I removed the space where the bio should be if there is no bio in the profile page. I also changed the colors of the 3 buttons in the profile page.

I know its a lot of changes. Please let me know if you find any errors or if something looks bad.

### CODE REVIEW

File: lib/view/pages/events/view_events_page.dart
Comment (MonikaTheiss) [inline lib/view/pages/events/view_events_page.dart:103]: I'm sorry I have to point this out again. Could you please leave in my code here ? 
This is what my code does:
Stream gets data everytime something changes on the database. Then I empty allEvents , so that allEvents always has the most up to date data from the stream. Then I put in the newest data from the stream into the allEvents object. I don't feel a check if allevents == null does the same. The way you did it means that the data stys the same until you refresh. My way always got the most up to date data when something changed, yes there are some weird things that don't immediatelly update ( I don't know why ) but a lot of things do update. Please put my code back in here.

File: lib/view/pages/game_library/game_library_page.dart
Comment (MonikaTheiss) [inline lib/view/pages/game_library/game_library_page.dart:142]: Small thing , could we perhaps leave these as Games and Connections, as this was something our mentors mentioned?

File: lib/view/pages/game_library/game_library_page.dart
Comment (MonikaTheiss) [inline lib/view/pages/game_library/game_library_page.dart:213]: And perhaps leave this in as well? Because there was an overflow issue or something and I asked @pieryvdl to fix it I think

Comment (u22520016) [reply lib/view/pages/game_library/game_library_page.dart:213]: I tried to remove this, but it broke the whole page...

File: lib/view/pages/game_library/game_library_page.dart
Comment (u22520016) [reply lib/view/pages/game_library/game_library_page.dart:142]: I don't know what you mean here.  It's already like that

File: lib/view/pages/events/view_events_page.dart
Comment (u22520016) [reply lib/view/pages/events/view_events_page.dart]: Fixed

File: lib/view/pages/game_library/game_library_page.dart
Comment (MonikaTheiss) [reply lib/view/pages/game_library/game_library_page.dart:213]: what exactly broke?

File: lib/view/pages/game_library/game_library_page.dart
Comment (MonikaTheiss) [reply lib/view/pages/game_library/game_library_page.dart:142]: It should be Connections and Games , not CONNECTIONS and GAMES

Comment (u22520016) [reply lib/view/pages/game_library/game_library_page.dart:142]: Hmm, in my code its already like that

File: lib/view/pages/game_library/game_library_page.dart
Comment (u22520016) [reply lib/view/pages/game_library/game_library_page.dart:213]: The page is empty, which means something has than infinite height

**Review by MonikaTheiss — CHANGES_REQUESTED** [review]
Sent you a whatsapp messages. My only problem right now is the overflows that pop up. Everything else looks great !

**Comment by pieryvdl** [comment]
Just waiting for Monika's approval. I can't fault anything on my side. Looks good Thanks Franco!

**Comment by MonikaTheiss** [comment]
@shan-226 could you perhaps run this on your phone to check for any other missed overflows please?

**Comment by shan-226** [comment]
> @shan-226 could you perhaps run this on your phone to check for any other missed overflows please?

The only overflow on this branch that I could find is the one on the Help page that should already be fixed and in develop. 
Other than that everything looks good to me.

**Comment by shan-226** [comment]
> @shan-226 could you perhaps check the overflow on the play button again please?
It's gone. No more overflow!

**Review by shan-226 — APPROVED** [review]

File: lib/view/pages/events/view_events_page.dart
Comment (pieryvdl) [inline lib/view/pages/events/view_events_page.dart:25]: Can you please reinstate the badge service lines?

File: lib/view/pages/events/view_events_page.dart
Comment (pieryvdl) [inline lib/view/pages/events/view_events_page.dart:39]: Here as well. Needed for the badge service.

File: lib/view/pages/game_library/game_library_page.dart
Comment (pieryvdl) [inline lib/view/pages/game_library/game_library_page.dart:39]: Needed for badge service. Please 😄

File: lib/view/pages/game_library/game_library_page.dart
Comment (pieryvdl) [inline lib/view/pages/game_library/game_library_page.dart:121]: Need this badgeService as well please.

File: lib/view/pages/home/home_page.dart
Comment (pieryvdl) [inline lib/view/pages/home/home_page.dart:38]: This too please.

File: lib/view/pages/home/home_page.dart
Comment (pieryvdl) [inline lib/view/pages/home/home_page.dart:68]: Need these 2 lines too. 😄

**Review by pieryvdl — COMMENTED** [review]
Rest is all good thank you. Will approve when those lines are back. 😄

**Review by pieryvdl — APPROVED** [review]

**Pull request merged**
